engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -726,16 +778,40 @@ namespace dmGameObject
 
         dmGameObject::PropertyOptions property_options;
         property_options.m_Index = 0;
+        property_options.m_HasKey = 0;
 
         // Options table
         if (lua_gettop(L) > 3)
         {
             luaL_checktype(L, 4, LUA_TTABLE);
             lua_pushvalue(L, 4);
 
+            lua_getfield(L, -1, "key");
+            if (!lua_isnil(L, -1))
+            {
+                if (lua_isstring(L, -1))
+                {
+                    property_options.m_Key = dmHashString64(lua_tostring(L, -1));
+                }
+                else if (dmScript::IsHash(L, -1))
+                {
+                    property_options.m_Key = dmScript::CheckHash(L, -1);
+                }
+                else
+                {
+                    return luaL_error(L, "Invalid key argument in options table. It should be string or hash.");
+                }
```

Comment:
Same as above, CheckHashOrString() should suffice

Reply:
👍